sympy/core/add.py

```diff
@@ -263,6 +263,12 @@ def flatten(cls, seq):
                 # x + O(x**2) -> x + O(x**2)
                 if t is not None:
                     newseq2.append(t)
+
+            # 2 + O(0) -> 2
+            for o in order_factors:
+                if o.expr.is_zero:
+                    order_factors.remove(o)
```

Comment:
You shouldn't modify a list while looping through it in Python.

Can this loop be merged with the one above?

Reply:
No, it can be merged with the loop above because if newseq is empty it doesn't iterate through the loop.
Btw thanks for pointing out the fact one shouldn't modify a list while looping through it.